src/components/parameters/component-parameters-implementation.adb

```diff
@@ -361,9 +376,66 @@ package body Component.Parameters.Implementation is
             Value (Value'First .. Value'First + Param_Length - 1) := To_Byte_Array (Ptr);
 
             -- Stage the parameter:
-            Set_Status (Self.Stage_Parameter (Param_Entry => Param_Entry, Value => Value));
+            if Self.Stage_Parameter (Param_Entry => Param_Entry, Value => Value) /= Success then
+               Status_To_Return := Parameter_Error;
+            end if;
          end;
       end loop;
+      return Status_To_Return;
+   end Stage_Parameter_Table;
+
+   -- Helper function to update all connected component's parameter tables from data within a provided memory region.
+   function Validate_Parameter_Table (Self : in out Instance; Region : in Memory_Region.T) return Parameters_Memory_Region_Release.T is
+      use Parameter_Enums.Parameter_Update_Status;
+      use Parameter_Enums.Parameter_Table_Update_Status;
+      Status_To_Return : Parameter_Enums.Parameter_Table_Update_Status.E := Success;
+   begin
+      -- Send info event:
+      Self.Event_T_Send_If_Connected (Self.Events.Starting_Parameter_Table_Validate (Self.Sys_Time_T_Get, Region));
+
+      -- Stage all of the parameters:
+      Status_To_Return := Self.Stage_Parameter_Table (Region);
+
+      -- OK, now we need to validate all the parameters.
+      for Idx in Self.Connector_Parameter_Update_T_Provide'Range loop
+         if Self.Is_Parameter_Update_T_Provide_Connected (Idx) then
```

Comment:
It is weird to me that we check for if connected here. We also check in Update_Parameter_Table. But I don't think there is a commensurate check for staging parameters. So we would fail to stage if we were disconnected. I think to make this uniform we should if an if_connected check to the stage as well. And also to the fetch if that is not implemented in that way.

Reply:
Added similar connection checks to `Stage_Parameter` and `Fetch_Parameters` in [add37f4](https://github.com/lasp/adamant/pull/65/commits/add37f417cd827f5e74a404f840b9652b210d099).